core/src/main/java/org/testcontainers/lifecycle/Startables.java

```diff
@@ -72,6 +73,18 @@ public Thread newThread(Runnable r) {
             })
             .toArray(CompletableFuture[]::new);
 
-        return CompletableFuture.allOf(futures);
+        return allOfShortcircuiting(futures);
+    }
+
+    private static CompletableFuture<Void> allOfShortcircuiting(CompletableFuture[] futures) {
+        CompletableFuture<Void> result = new CompletableFuture<>();
+
+        for (CompletableFuture<?> f : futures) {
+            f.handle((__, ex) -> ex == null || result.completeExceptionally(ex));
```

Comment:
Although this construction is valid, could you please replace it with `whenComplete` and `if (ex != null)`?